src/DateTimePicker.jsx

```diff
@@ -223,7 +223,7 @@ var DateTimePicker = React.createClass({
               disabled={disabledOrReadonly}
               aria-disabled={disabledOrReadonly}
               aria-label={messages.calendarButton}
-              onClick={() => this.toggle(popups.CALENDAR)}
+              onClick={tetherPopup ? () => this.toggle(popups.CALENDAR) : this._click.bind(null, popups.CALENDAR)}
```

Comment:
Revises `else` condition to exact previous state: https://github.com/procore/react-widgets/blob/3be1b7dd5835c4d54bb8366bbb23ccaf181a90ca/src/DateTimePicker.jsx#L218